### PR TITLE
Add precommit

### PR DESCRIPTION
**Pull request by deanwilson**
Add terraform precommit hooks for this repo and clean up the issues they raise.

### CODE REVIEW

File: providers.tf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+provider "aws" {
+  region  = "${var.aws_region}"
+  version = "1.30.0"
```
Comment (netflash) [inline providers.tf]: ~> 1.30.0

File: providers.tf [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+}
+
+provider "template" {
+  version = "1.0.0"
```
Comment (netflash) [inline providers.tf]: ~> 1.0.0

**Review by netflash — APPROVED** [review]
/lgtm

File: providers.tf (hunk 1 above)
Comment (deanwilson) [reply providers.tf]: Fixed. Thanks!

**Pull request merged**